sklearn/decomposition/fastica_.py

```diff
@@ -350,6 +356,7 @@ def g(x, fun_args):
     if whiten:
         if compute_sources:
             S = np.dot(np.dot(W, K), X).T
+            S /= np.sqrt(np.var(S))
```

Comment:
S should have shape (n_samples, n_components) here. `np.var(S)` will compute the variance over all sources (components) and samples rather than for each source across samples.
```suggestion
            S_std = np.std(S, axis=0, keepdims=True)
            S /= S_std
```

Then, to make this equation consistent again
```python
S = np.dot(np.dot(W, K), X).T
```
you'll need to modify the right side by modifying W
```python
W /= S_std.T
```